docs/proposal_drafts/.draft_template.md

```diff
@@ -0,0 +1,48 @@
+# Introduction
+
+## `Problem` in short
+
+<!--
+Brief introduction, focus on clearly expressing the issue!
+-->
+
+
+## `Problem` detailed
+
+<!--
+Clearly details the issue you are trying to solve.
+Include any technical details or specifications necessary for understanding the solution.
+-->
+
+---
+
+# Constraints
+
+## Desired `outcome`
+
+<!--
+Clearly express the desired outcome or goal.
+-->
+
+## List of `edge cases`
+
+<!--
+Solutions are not perfect and usually fail in some situation.
+Highlight `likely` and `noteworthy` edge cases.
+-->
+
+---
+
+# Proposal
+
+## `Solution` in short
+
+<!--
+Provide a high-level overview of how the proposed solution will be implemented.
+-->
+
+## `Solution` detailed
+
+<!--
```

Comment:
How exactly would you use this document? Is this like some kind of draft to start a discussion based on, or is this supposed to remain in the codebase?
I think it is likely that these kind of documents would deviate from the actual implementation over time, but I think they could be useful to "draft" a solution with other developers before actually starting the implementation and they might be able to spot flaws/improve the design before implementing. In that way it could be useful I think.

Reply:
I personally find well structured user stories to be a great help, especially in the front-end.

This is a decent read and example. [https://www.atlassian.com/agile/project-management/user-stories](https://www.atlassian.com/agile/project-management/user-stories)

I sometimes think we are missing a little the refinement of stories to address exactly what you are talking about

Reply:
The idea here is to have a `draft` that clearly points out a problem that a developer is facing.
- I know and agree that this will fall out of date as the codebase evolves, but it will still be kept as historic proof/fact of how this came into action.
- And the usage would be: Hi people I have this issue and this is how I'd like to solve it. What do you think? Is there a better way to do it?

The idea is to start a discussion.

Reply:
@jsaq007 please keep in mind that this is not a feature request. It's a way to create a design draft for a new feature (sometimes these features are had to explain).